### DISCUSSION TITLE
RIC metrics list

### DISCUSSION THREAD

**Discussion by fllay** [srsran/srsRAN_Project#210] [General Help] [answered]
Now only three metrics are exposed. May I ask what are all  metrics for E2SM_KPM?  Is there any standard about metrics for E2SM_KPM?  Or it depends on gNB implementation. For example, will PRB and numer of UEs  be implemented? Are there mandatory metrics that have to implement?  I tried to find any information about this but I can find only few. Thank you.

**Comment by yagoda** [comment, marked as answer]
Hi,
The metrics are listed in two different places:
-  3GPP TS 28.552
- O-RAN.WG3.E2SM-KPM-R003-v3.00
We will be expanding the list of supported metrics in later releases according to demand, the three metrics currently implemented were meant to serve as an example for anyone who wished to implement their own metrics.

If you have specific metrics from the above documents you are interested in, please share them here and we will take them into consideration!

Thanks,
Justin

**Reply by mrouili** [reply]
Hello @yagoda, are there any instructions or guidelines/tutorials on how would one go about implementing their own metrics? 
Thank you !

**Reply by fllay** [reply]
Thank you for clarification. I am interested in Section 5.1.1, Performance measurements valid for all gNB deployment scenarios, of  3GPP TS 28.552. Especially, radio resource metrics  of Section 5.1.1. It would be nice if there is a document about how to implement our own metrics.

**Reply by pgawlowicz** [reply]
Currently, we do not have such guidelines, but you can have a look and check how the currently supported metrics are implemented. Basically, [e2sm_kpm_du_meas_provider_impl](https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_du_meas_provider_impl.cpp) is an objects that translates the internal gnb metrics into the E2 metrics.
